### ISSUE TITLE
[bug] `naming-convention/component-name` rule does not work in some cases

### ISSUE THREAD

**Issue by 13OnTheCode** [Rel1cx/eslint-react#734]
### Describe the bug

Hi there,

I've encountered an issue with the `naming-convention/component-name` rule. It seems like it's not correctly recognizing components that follow the PascalCase naming convention.

```ts
export function test() { // Expected to be wrong, but passed
  return (
    <div />
  )
}

export function TEST() { // Expected to be wrong, but passed
  return (
    <div />
  )
}

export function _Test() { // Expected to be wrong, but passed
  return (
    <div />
  )
}

export function Test_() {
  //     ^^^^^^^^^^^^^^^^^
  //     - Expected component name to be in 'PascalCase'.
  return (
    <div />
  )
}

export function Test_Test() {
  //     ^^^^^^^^^^^^^^^^^
  //     - Expected component name to be in 'PascalCase'.
  return (
    <div />
  )
}
```

### Platform and versions

```Text
-
```

**Comment by 13OnTheCode** [comment]
@Rel1cx  Hi there,

I noticed the latest update fixed this, but by default, it still doesn't flag camelCase or components with a leading underscore as errors.
 
 ```jsx
/* eslint react/naming-convention/component-name: [2, "PascalCase"] */

export function test() { // Expected to be wrong, but passed
  return (
    <div />
  )
}

export function _Test() { // Expected to be wrong, but passed
  return (
    <div />
  )
}
```

**Comment by Rel1cx** [comment]
> https://github.com/Rel1cx/eslint-react/blob/fd4295289572a4e7511ded2fc0f3e80c58b350d5/packages/plugins/eslint-plugin-react-naming-convention/src/rules/component-name.spec.ts#L48-L49
> 
> @Rel1cx Why are `testcomponent` and `testComponent` considered valid PascalCase names? According to PascalCase conventions, each word should start with an uppercase letter and there shouldn’t be any separators. I noticed the update didn’t fix camelCase component names. Is there a specific reason for this?

This is a completely legitimate usage, because in React, all components starting with a lowercase letter are treated by React as `host components` and are created internally by the `React Renderer`, which distinguishes them from `user-defined components`.

It is recommended to understand the JSX specification and how `react-reconciler` works.

**Comment by 13OnTheCode** [comment]
> > https://github.com/Rel1cx/eslint-react/blob/fd4295289572a4e7511ded2fc0f3e80c58b350d5/packages/plugins/eslint-plugin-react-naming-convention/src/rules/component-name.spec.ts#L48-L49
> > 
> > @Rel1cx Why are `testcomponent` and `testComponent` considered valid PascalCase names? According to PascalCase conventions, each word should start with an uppercase letter and there shouldn’t be any separators. I noticed the update didn’t fix camelCase component names. Is there a specific reason for this?
> 
> This is a completely legitimate usage, because in React, all components starting with a lowercase letter are treated by React as `host components` and are created internally by the `React Renderer`, which distinguishes them from `user-defined components`.
> 
> It is recommended to understand the JSX specification and how `react-reconciler` works.

Oh, I see, I misunderstood. I thought this rule was meant to enforce naming conventions when users declare components. Thanks for taking the time to explain this, and I really appreciate your contribution!
